.circleci/config.yml

```
@@ -170,22 +194,36 @@ workflows:
           filters:
             tags:
               only: /.*/
-      - docker-build:
+      - setup-env:
           requires:
             - frontend-services-checks
             - backend-services-checks
           filters:
             tags:
               only: /.*/
+      - docker-build-backend:
+          requires:
+            - setup-env
```

This is good. Can make it dependent to frontend-services-checks and backend-services-checks as well
```suggestion
            - setup-env
            - frontend-services-checks
            - backend-services-checks
```

---

It is a chain, first the checks complete only then the `setup-env` works so we don't need another dependency here. You can take a look at the current build to see how things are dependant right now.

---

After you add these, all three of these run parallel, which reduces extra the setup-env time